libs/pika/execution/tests/unit/algorithm_when_all_vector.cpp

```diff
@@ -0,0 +1,218 @@
+    }
+
+    // Failure path
+    {
+        std::atomic<bool> set_error_called{false};
+        auto s = ex::when_all_vector(std::vector{error_typed_sender<double>{}});
+        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
+            check_exception_ptr, set_error_called};
+        auto os = ex::connect(std::move(s), std::move(r));
+        ex::start(os);
+        PIKA_TEST(set_error_called);
+    }
+
+    {
+        std::atomic<bool> set_error_called{false};
+        std::vector<ex::unique_any_sender<double>> senders;
+        senders.emplace_back(error_typed_sender<double>{});
+        senders.emplace_back(ex::just(42.0));
+        senders.emplace_back(ex::just(43.0));
+        senders.emplace_back(ex::just(44.0));
+        auto s = ex::when_all_vector(std::move(senders));
+        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
+            check_exception_ptr, set_error_called};
+        auto os = ex::connect(std::move(s), std::move(r));
+        ex::start(os);
+        PIKA_TEST(set_error_called);
+    }
+
+    {
+        std::atomic<bool> set_error_called{false};
+        std::vector<ex::unique_any_sender<double>> senders;
+        senders.emplace_back(error_typed_sender<double>{});
+        senders.emplace_back(ex::just(42.0));
+        senders.emplace_back(ex::just(43.0));
+        senders.emplace_back(ex::just(44.0));
+        auto s = ex::when_all_vector(std::move(senders));
+        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
+            check_exception_ptr, set_error_called};
+        auto os = ex::connect(std::move(s), std::move(r));
+        ex::start(os);
+        PIKA_TEST(set_error_called);
+    }
+
+    {
+        std::atomic<bool> set_error_called{false};
+        std::vector<ex::any_sender<double>> senders;
+        senders.emplace_back(error_typed_sender<double>{});
+        senders.emplace_back(ex::just(42.0));
+        senders.emplace_back(ex::just(43.0));
+        senders.emplace_back(ex::just(44.0));
+        auto s = ex::when_all_vector(std::move(senders));
+        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
+            check_exception_ptr, set_error_called};
+        auto os = ex::connect(std::move(s), std::move(r));
+        ex::start(os);
+        PIKA_TEST(set_error_called);
+    }
+
+    {
+        std::atomic<bool> set_error_called{false};
+        std::vector<ex::any_sender<double>> senders;
+        senders.emplace_back(error_typed_sender<double>{});
+        senders.emplace_back(ex::just(42.0));
+        senders.emplace_back(ex::just(43.0));
+        senders.emplace_back(ex::just(44.0));
+        auto s = ex::when_all_vector(std::move(senders));
+        auto r = error_callback_receiver<decltype(check_exception_ptr)>{
+            check_exception_ptr, set_error_called};
+        auto os = ex::connect(std::move(s), std::move(r));
+        ex::start(os);
+        PIKA_TEST(set_error_called);
+    }
```

Comment:
Should we also add a test with a sender that is not just (like with a longer task) or does it not make sense?

Reply:
I added some tests that actually spawn tasks to the `thread_pool_scheduler` test over here: https://github.com/pika-org/pika/blob/d439f181006f3358500390740680ea053b5dfa6c/libs/pika/executors/tests/unit/thread_pool_scheduler.cpp#L661-L774. Do you think that covers it?